lib/foreman_discovery/engine.rb

```diff
@@ -19,6 +19,19 @@ class Engine < ::Rails::Engine
       require_dependency File.expand_path("../../../app/models/setting/discovered.rb", __FILE__) if (Setting.table_exists? rescue(false))
     end
 
+    initializer 'foreman_discovery.assets.precompile' do |app|
+      app.config.assets.precompile += %w(
+        'discovered_hosts.js',
+      )
+    end
+
+    initializer 'foreman_discovery.configure_assets', :group => :assets do
+      SETTINGS[:foreman_discovery] =
+          {:assets => {:precompile => [
+              'discovered_hosts.js',
+          ]}}
+    end
+
```

Comment:
@lzap i can do it inline so we won't have to do a repackaging.
About the feature as a whole i haven't finished yet, this was just the proto.